examples/layer2/pre_fabric.yml

```
@@ -0,0 +1,102 @@
+---
+- name: Equinix Metal Example Playbook
+  hosts: localhost
+  gather_facts: no
+  tasks:
+    - name: Include the required variables
+      include_vars: "vars/vars.yml"
+
+    # Equinix resources
+    - name: Create a project
+      equinix.cloud.metal_project:
+        name: "{{ project_name }}"
+      register: project
+
+    - name: create a vlan
+      equinix.cloud.metal_vlan:
+        project_id: "{{ project.id }}"
+        metro: "{{ metro }}"
+        vxlan: "1234"
+      register: vlan
+
+    - name: create a VRF
+      equinix.cloud.metal_vrf:
+        name: "{{ vrf_name }}"
+        metro: "{{ metro }}"
+        local_asn: "{{ metal_side_asn }}"
+        ip_ranges:
+          - "{{ vrf_peering_ip_range }}"
+          - "{{ vrf_gateway_ip_range }}"
+        project_id: "{{ project.id }}"
+      register: vrf
+
+    - name: create a VRF IP reservation
```

```suggestion
    - name: configure BGP for interconnection virtual circuit
      equinix.cloud.metal_virtual_circuit:
        id: "{{ connection.ports[0].virtual_circuits[0].id }}"
        peer_asn: "{{ vrf_peering_asn }}"
        customer_ip: "{{ aws_peering_ip }}"
        metal_ip: "{{ metal_peering_ip }}"
        subnet: "{{ vrf_vc_peering_ip_range }}"
        md5: "{{ bgp_md5_password }}"
        # The metal_virtual_circuit module requires this parameter
        # in order to know that the circuit is a VRF circuit and
        # not a VLAN circuit
        vrf: "{{ vrf.id }}"

```

We still need to create the BGP configuration on the virtual circuit.